File: 프로그래머스/lv1/42840. 모의고사/모의고사.js
function solution(answers) {
  const mathGiveUper1 = [1, 2, 3, 4, 5];
  const mathGiveUper2 = [2, 1, 2, 3, 2, 4, 2, 5];
  const mathGiveUper3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];

  let answer1 = 0;
  let answer2 = 0;
  let answer3 = 0;
  for (let i = 0; i < answers.length; i++) {
    if (answers[i] === mathGiveUper1[i % 5]) {
      answer1++;
    }
    if (answers[i] === mathGiveUper2[i % 8]) {
      answer2++;
    }
    if (answers[i] === mathGiveUper3[i % 10]) {
      answer3++;
    }
  }
  const maxValue = Math.max(answer1, answer2, answer3);

  const ans = [];
  if (answer1 === maxValue) ans.push(1);
  if (answer2 === maxValue) ans.push(2);
  if (answer3 === maxValue) ans.push(3);
  return ans;
}

File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = [0, 0, 0]
    n = len(answers)
    answer1 = [1, 2, 3, 4, 5]
    answer2 = [2, 1, 2, 3, 2, 4, 2, 5]
    answer3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    for i in range(len(answers)):
        if answers[i] == answer1[i % 5]:
            answer[0] += 1
        if answers[i] == answer2[i% 8]:
            answer[1] += 1
        if answers[i] == answer3[i % 10]:
            answer[2] += 1
    
    maxValue = max(answer)
    result = [i + 1 for i in range(3) if answer[i] == maxValue]
    return result